fix: search example to support Redis v4+ and Express 4/5

vinybrun opened the pull request:
This PR updates the Search example to address issues caused by breaking changes in Redis v4+ and Express 5. 

Key changes include:
- Replacing `sadd` with `sAdd` and `smembers` with `sMembers` for Redis v4+ compatibility.
- Using `{0,1}` syntax for optional parameters instead of `?` to support Express 5.
- Adding error handling for Redis connection and commands.

These changes ensure the example works seamlessly.

Tested locally with:
Redis 7.0.15
Node.js 22.12.0
Express 5.0.1 and 4.21.2
Debian 12

Note:
Adding automated tests for this example could help catch issues like this sooner. If you think it's worthwhile, I’d be glad to help implement them.

<!--
Thank you for your pull request. Please provide a description and 
note the Certificate of Origin below. 

-->

<!--
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

vinybrun committed:
Fix Redis example to support Redis v4+ and Express 4/5

wesleytodd commented on examples/search/index.js:
```diff
@@ -19,28 +19,41 @@ var db = redis.createClient();
-    if (err) return next(err);
-    res.send(vals);
-  });
+app.get('/search/:query{0,1}', function (req, res, next) {
```
This is probably better done as `/{:query}` if we intend it to be optional like the original example.

vinybrun replied:
Thanks for pointing this out! I've updated the route accordingly.

wesleytodd commented on examples/search/index.js:
```diff
@@ -19,28 +19,41 @@ var db = redis.createClient();
+  } catch (err) {
+    console.error('Error populating Redis:', err);
+  }
+})();
```
Oh oops, I forgot to hit add on this one. I had tried to comment that this change led to some unexpected ordering issues. My original comment was likely better, but for this I think we should make sure that these being run after the server starts is not an issue (because they will be since this is not `aiwat`/`.then`'d before the server starts.

vinybrun replied:
Appreciate the input! I've refactored the redis setup into a dedicated function `initializeRedis` and called it with `await` before the server starts.

vinybrun committed:
update optional route syntax to /{:query} and refactor Redis initialization into dedicated function to guarantee that it is complete before server starts

vinybrun commented:
Hi @wesleytodd, I've made the requested changes: 
- Updated optional route syntax to `/{:query}`.
- Refactored Redis initialization into a dedicated function  `initializeRedis` to make sure that it completes before the server starts. 

I think this aligns with your feedback, but please let me know if I misunderstood or missed anything. Thanks!

wesleytodd reviewed (approved):
Thanks for addressing the feedback. LGTM!

bjohansebas committed:
Merge branch 'master' of github.com:expressjs/express into fix-redis-example

bjohansebas reviewed (approved):
LGTM. I don’t think we want to add more devDependencies, and it seems it was never the intention to add them, those were just examples.
